app/api/channels/route.ts

```
@@ -1,32 +1,50 @@
   const channelId = searchParams.get('channelId');
-  const inProgress = searchParams.get('inProgress');
+  const channelType = searchParams.get('channelType');
+
+  let rawLimit = searchParams.get('limit');
+  let rawOffset = searchParams.get('offset');
+  if (!channelId && (rawLimit === null || rawOffset === null)) {
+    return NextResponse.json({ error: 'Invalid request' }, { status: 400 });
+  }
+  const limit = Number(rawLimit);
+  const offset = Number(rawOffset);
+  if (isNaN(limit) || isNaN(offset)) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)